mula/scheduler/schedulers/boefje.py

```diff
@@ -997,3 +998,50 @@ def get_boefjes_for_ooi(self, ooi) -> list[Plugin]:
         )
 
         return boefjes
+
+    def post_push(self, item: Task) -> Task:
+        """Override Schedule.post_push() when a boefje specifies a schedule for
+        execution (cron expression) we schedule for its execution"""
+        # Does a boefje have a schedule defined?
+        plugin = self.ctx.services.katalogus.get_plugin_by_id_and_org_id(
+            utils.deep_get(item.data, ["boefje", "id"]),
+            self.organisation.id,
+        )
+        if plugin is None or plugin.cron is None:
+            return super().post_push(item)
```

Comment:
Flow of control is a bit confusing here: if we have no `cron`, we let the parent handle the item, but there are checks like `if self.create_schedule` there that we don't do here, and the other important field being the `interval` here is handled because we also override the `calculate_deadline` method.

In short I feel like we should be able to override smaller methods here, perhaps override a `create_cron(item)` method so that you don't have to override the `post_push`?

Reply:
Yeah I agree, during our meeting it also was difficult to convey the purpose and why the overrides were necessary. I like your suggestion, and will try update it to make easier to reason about.